### PR TITLE
Feature/api dataclass

### PR DESCRIPTION
**Pull request by saeliddp**
## What does this PR do?
Add importable enums in `api/data.py` for a few config items with a restricted set of acceptable values. Most importantly, exposes auto-thresholding methods, which can then be used to update the new target in Benji's PR. Also added some unit tests for a corresponding modification I made to `CytoDLModel`.

See [this doc](https://alleninstitute-my.sharepoint.com/:w:/g/personal/daniel_saelid_alleninstitute_org/EaRGLGO2zgRDmAC8nX5ChocBcxePXOTnlcHQxODIbaVA8g?e=99ASMS) for information about the Cyto-DL API expansion (and please comment/edit if you have any suggestions or concerns!).

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?
- [x] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: cyto_dl/api/data.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    # other hardware types available, but require more complicated config
+
+
+class PatchSize(Enum):
+    """Patch size for training, and their respective patch shapes.
+
+    TODO: get from benji
+    """
+
+    SMALL = [1, 3, 3]
```
Comment (benjijamorris) [inline cyto_dl/api/data.py:28]: Brian and I talked about updating these to 
small = [16, 32, 32]
medium = [16, 64, 64]
large = [16, 128, 128]

File: cyto_dl/api/data.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    MEDIUM = [16, 32, 32]
+    LARGE = [20, 40, 40]
+
+
+# importing skimage takes a while.
+# could speed it up by hardcoding this enum, but would have to update
+# if skimage adds/deletes threshold filters
+
+AutoThresholdMethod = Enum(
```
Comment (benjijamorris) [inline cyto_dl/api/data.py:37]: awesome!

File: cyto_dl/api/model.py [hunk 3]
```diff
@@ -45,19 +44,21 @@ def load_config_from_dict(self, config: dict):
         """Load configuration from dictionary."""
         self.cfg = config
 
+    # TODO: replace experiment_type str with api.data.ExperimentType -> will
```
Comment (benjijamorris) [inline cyto_dl/api/model.py:47]: done!

**Review by benjijamorris — DISMISSED** [review]
This looks great Danny!

File: cyto_dl/api/data.py (hunk 1 above)
Comment (saeliddp) [reply cyto_dl/api/data.py]: Done ([commmit](https://github.com/AllenCellModeling/cyto-dl/pull/347/commits/0e3828a151966ff1ae2550a4db25cdf285382d02))

File: cyto_dl/api/data.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "AutoThresholdMethod",
+    {
+        func_name.split("threshold_")[-1].upper(): func_name
+        for func_name in filter(lambda x: x.startswith("threshold_"), dir(skimage.filters))
```
Comment (hughes036) [inline cyto_dl/api/data.py:38]: Is `skimage` pinned?  I imagine that `skimage.filters` could change and break assumpions in this code

Comment (hughes036) [reply cyto_dl/api/data.py:38]: I dont think i understand how `AutoThresholdMethod` is used - is it exercised by `model.load_default_experiment` (and called by the new tests)?

Comment (hughes036) [reply cyto_dl/api/data.py:38]: Ok, i get it after seeing the example in your [doc](https://alleninstitute-my.sharepoint.com/:w:/g/personal/daniel_saelid_alleninstitute_org/EaRGLGO2zgRDmAC8nX5ChocBcxePXOTnlcHQxODIbaVA8g?e=99ASMS).  How about including a docstring with that usage example?

**Review by hughes036 — APPROVED** [review]

**Pull request merged**